### ISSUE TITLE
Add more notes on specificity to "Existing CSS"

### ISSUE THREAD

**Issue by philpl** [styled-components/styled-components-website#177]
https://www.styled-components.com/docs/advanced#existing-css

In this section we give some examples on how to edit the exisiting CSS files / code to override styled-components' styling, but it turned out that the reverse is more popular. So a lot of issues are being opened repeatedly, asking on how to make styled-components generate classes with a higher-specificity.

An example issue: https://github.com/styled-components/styled-components/issues/1253

Basically this should be illustrated in the section as well:

```js
const StyledTypography = styled(Typography)`
  && {
    color: palevioletred;
    font-weight: bold;
  }
`;
```

**Comment by adamchenwei** [comment]
This is really a hack, I wouldn't call it a long-term solution, because the nature of it. It not as bad as using !important though lol

**Comment by morajabi** [comment]
@adamchenwei When the user is using SC on an existing codebase with another styling solution which who knows how that works (!) the only we to deal with that is this. Adding SC on top of that codebase is "a hack".

**Comment by philpl** [comment]
@adamchenwei Unfortunately it's the only solution when dealing with existing CSS. I don't see a solution to this as a "first citizen" of styled-components, as it's simply a best practice-breaking solution to introduce external CSS in the first place. Thus this workaround is here to stay and will need to be documented to prevent any further confusion.

I can see a quick babel plugin as a solution (which I suggested in an issue at some point) that adds n levels of specificity to each styled component.

So, yea, it doesn't classify as a hack, but as a workaround, in my opinion.

**Comment by mxstbr** [comment]
This is now in the FAQ.
